internal/data/generated_trace_test.go

```diff
@@ -51,6 +176,81 @@ func TestSpanEvent(t *testing.T) {
 	assert.EqualValues(t, generateTestSpanEvent(), ms)
 }
 
+func TestSpanLinkSlice(t *testing.T) {
+	es := NewSpanLinkSlice(0)
+	assert.EqualValues(t, 0, es.Len())
+	es = newSpanLinkSlice(&[]*otlptrace.Span_Link{})
+	assert.EqualValues(t, 0, es.Len())
+	es = NewSpanLinkSlice(13)
+	defaultVal := NewSpanLink()
+	testVal := generateTestSpanLink()
+	assert.EqualValues(t, 13, es.Len())
+	for i := 0; i < es.Len(); i++ {
+		assert.EqualValues(t, defaultVal, es.Get(i))
+		fillTestSpanLink(es.Get(i))
+		assert.EqualValues(t, testVal, es.Get(i))
+	}
+
+	// Test resize.
+	const resizeLo = 2
+	const resizeHi = 10
+	expectedEs := make(map[*otlptrace.Span_Link]bool, resizeHi-resizeLo)
+	for i := resizeLo; i < resizeHi; i++ {
+		expectedEs[es.Get(i).orig] = true
+	}
+	assert.EqualValues(t, resizeHi-resizeLo, len(expectedEs))
+	es.Resize(resizeLo, resizeHi)
+	assert.EqualValues(t, resizeHi-resizeLo, es.Len())
+	foundEs := make(map[*otlptrace.Span_Link]bool, resizeHi-resizeLo)
+	for i := 0; i < es.Len(); i++ {
+		foundEs[es.Get(i).orig] = true
+	}
+	assert.EqualValues(t, expectedEs, foundEs)
+
+	// Test remove.
+	const removePos = 2
+	delete(expectedEs, es.Get(removePos).orig)
+	es.Remove(removePos)
+	assert.EqualValues(t, resizeHi-resizeLo-1, es.Len())
+	foundEs = make(map[*otlptrace.Span_Link]bool, resizeHi-resizeLo)
+	for i := 0; i < es.Len(); i++ {
+		foundEs[es.Get(i).orig] = true
+	}
+	assert.EqualValues(t, expectedEs, foundEs)
```

Comment:
Would be nice to check the actual indexes instead of the presence. For example, change `foundEs` to be a `map[*otlptrace.Span_Link]int` and store indexes in it and then verify after removing.

Reply:
I would not test a specific behavior, currently I move the last element instead of shifting all the elements, unless you want to guarantee a specific behavior.